edb/server/compiler/rpc.pyx

```diff
@@ -210,17 +239,44 @@ cdef class CompilationRequest:
         return (
             self.source.cache_key() == other.source.cache_key() and
             self.protocol_version == other.protocol_version and
+            self.input_language == other.input_language and
             self.output_format == other.output_format and
             self.input_format == other.input_format and
             self.expect_one == other.expect_one and
             self.implicit_limit == other.implicit_limit and
             self.inline_typeids == other.inline_typeids and
             self.inline_typenames == other.inline_typenames and
-            self.inline_objectids == other.inline_objectids
+            self.inline_objectids == other.inline_objectids and
+            self.role_name == other.role_name and
+            self.branch_name == other.branch_name
+        )
+
+
+cdef CompilationRequest _deserialize_comp_req(
+    data: bytes,
+    query_text: str,
+    compilation_config_serializer: sertypes.CompilationConfigSerializer,
+):
+    cdef:
+        ReadBuffer buf = ReadBuffer.new_message_parser(data)
+        CompilationRequest req
+
+    if data[0] == 1:
+        req = _deserialize_comp_req_v1(
+            buf, query_text, compilation_config_serializer)
+    else:
+        raise errors.UnsupportedProtocolVersionError(
+            f"unsupported compile cache: version {data[0]}"
         )
```

Comment:
I think this would fail in `_introspect_db()` -> `hydrate_cache()` after instance upgrade with v0 cache. Would it be possible to still load v0 cache entries and upgrade to v1 cache entries?

Reply:
This only applies to in-place upgrades and you can only upgrade in-place from a v1 cache instance no?

Reply:
ok I see